### PR TITLE
WIS-2660 | Add first-party subdomain support to gb-tracker-client cookie tracking

### PR DESCRIPTION
**Pull request by mattwelke**
Changes visitor ID cookie setting to use the apex domain when the hostname is detected to be a domain (aka not localhost). Requires testing by using on sites deployed to two domains.

### CODE REVIEW

File: test/gb-tracker-core.spec.ts [hunk 1]
```diff
@@ -360,7 +360,7 @@ describe('gb-tracker-core tests', () => {
     const window = new jsdom.JSDOM(undefined, { cookieJar }).window;
 
     const gbTrackerCore = new GbTracker('testcustomer', 'area');
-    const CookiesLib = gbTrackerCore.__getInternals().COOKIES_LIB;
+    const CookiesLib = gbTrackerCore.__getInternals().COOKIES_LIB as any;
```
Comment (mattwelke) [inline test/gb-tracker-core.spec.ts:363]: The new `as any`s are a result of me making the type stronger in the main code. The legacy test code mutated the cookies library that was imported, which is no longer valid.

File: src/utils/index.ts [hunk 2]
```diff
@@ -47,3 +47,23 @@ export function startsWithOneOf(target: string, array: string[]): boolean {
...
+export function getApexDomain(window: Window): string {
+    const host = window.location.hostname;
+
+    if (host.indexOf('.') < 0) {
+        // No . so do no split.
+        return host;
+    }
+
+    const split = host.split('.');
+    return `${split[split.length - 2]}.${split[split.length - 1]}`;
```
Comment (PeterPandelidis-GBI) [inline src/utils/index.ts:68]: How does this work for a co.uk domain

Comment (mattwelke) [reply src/utils/index.ts:68]: Good call. Will add more test cases and fix the source code.

**Comment by mattwelke** [comment]
Refactored the constructor so that the third param is an object instead of a string. Moved the existing logic for overriding the URL (which we use for testing) to this new options object. Updated docs too:

## Options

The constructor for the tracker client has a third, optional parameter for providing options:

```typescript
const tracker = new GbTracker('customer_id', 'area', {
    overrideUrl: '<some_url>' // Optional, overrides the URL the beacon is sent to. Useful for testing.
});
```

__________

Plan is to use this new parameter for all optional config moving forward, including the new cross-domain support. Functionality in this PR will do automatic cross-domain for all domains ending in `.com`. In the future, we'll add an option to this options param (as a non-breaking change) where people can put in the domains they use. This would override the automatic (for `.com` only) behavior for cross-domain.

**Review by SadatAhmed-GBI — APPROVED** [review]

**Pull request merged**